packages/grafana-ui/src/components/DataLinks/DataLinkInput.tsx

```diff
@@ -137,11 +137,10 @@ export const DataLinkInput: React.FC<DataLinkInputProps> = memo(
     }, []);
 
     const onVariableSelect = (item: VariableSuggestion, editor = editorRef.current!) => {
-      const includeDollarSign = Plain.serialize(editor.value).slice(-1) !== '$';
       if (item.origin !== VariableOrigin.Template || item.value === DataLinkBuiltInVars.includeVars) {
-        editor.insertText(`${includeDollarSign ? '$' : ''}\{${item.value}}`);
```

Comment:
This does seem alright to me as well, I cannot seem to find an answer to your question "Is there a scenario where you can trigger the variables suggestions inside the data-link editor without adding the $ character?"

The only thing related is [this in the docs](https://grafana.com/docs/grafana/latest/panels-visualizations/configure-data-links/#add-a-data-link) which says we can use Cmd+Space or Ctrl+Space to trigger the list of suggestions, but it doesn't work like that for me. I cannot find where this combination of keys is programatically caught and overwritten, if anywhere.

If the only way to open suggestions is through typing `$`, then this PR fixes the bugs 🙏😊.

Reply:
@polibb thanks so much for this review! so, weirdly, `ctrl` + `space` is triggering the suggestions menu on my mac + chrome (106.0.5249.119 (Official Build) (arm64))

also, as written this PR would totally mess up that functionality. I am thinking it might be better for me if I narrow the scope of this PR to just the template variable + double-dollarsign issue. I can re-use the `includeDollarSign` logic for the else statement that handles the template variables, then it will behave the same way as everything else without breaking anything existing.

would love to keep digging into this other, larger issue as well!

Reply:
It's curious this shortcut does nothing for me. Anyway, I'm glad it works for you.

Indeed, this is a good trade-off for now 😊. Thank you for tackling it! And for changing the description of the PR for easy view of what's happening.
> would love to keep digging into this other issue as well!

me too - we should probably be able to catch the last character behind the current cursor (?).